eos-tech-support/eos-write-live-image

```diff
@@ -0,0 +1,284 @@
+# can be used both as a live system, and to install itself. The image is
+# extracted to an exFAT partition, and is booted using a GRUB and
+# initramfs which know how to loopback-mount the image.
+#
+# NOTE: the partition table will not be in physical order so Windows can mount
+# the image partition. Additional images copied to the image partition will be
+# detected by the installer, but will not be bootable.
+
+EOS_WRITE_IMAGE=$(dirname $0)/eos-write-image
+if [ ! -f $EOS_WRITE_IMAGE ]; then
+    EOS_WRITE_IMAGE='eos-write-image'
+fi
+EOS_DOWNLOAD_IMAGE=$(dirname $0)/eos-download-image
+if [ ! -f $EOS_DOWNLOAD_IMAGE ]; then
+    EOS_DOWNLOAD_IMAGE='eos-download-image'
+fi
+
+ARGS=$(getopt -o o:x:lnmwfh -l "os-image:,windows-tool:,latest,ntfs,mbr,writable,force,help" -n "$0" -- "$@")
+eval set -- "$ARGS"
+
+usage() {
+    cat <<EOF
+Usage:
+   $0 [options] DEVICE
+
+Arguments:
+    DEVICE                 Device path (e.g. '/dev/sdb')
+
+Options:
+   -o,--os-image PATH      Path to Endless OS image
+   -x,--windows-tool PATH  Path to Endless OS installer tool for Windows
+   -l,--latest             Fetch latest OS image and/or Windows tool
+   -w,--writable           Allow image to be modified when run (which
+                           prevents installing it later)
+   -f,--force              don't ask to proceed before writing
+   -h,--help               Show this message
+
+Developer options (you probably don't want to use these):
+   -n,--ntfs               Format the image partition as NTFS, not exFAT
+   -m,--mbr                Format DEVICE as MBR, not GPT (implies --ntfs)
+EOF
+}
+
+function check_exists() {
+    if [ ! -f "$1" ]; then
+        echo "$2 $1 does not exist or is not a file" >&2
+        exit 1
+    fi
+}
+
+NTFS=false
+MBR=false
+WRITABLE=false
+FORCE=false
+while true; do
+    case "$1" in
+        -o|--os-image)
+            shift
+            OS_IMAGE="$1"
+            shift
+            ;;
+        -x|--windows-tool)
+            shift
+            WINDOWS_TOOL="$1"
+            shift
+            ;;
+        -l|--latest)
+            shift
+            FETCH_LATEST=true
+            ;;
+        -n|--ntfs)
+            shift
+            NTFS=true
+            ;;
+        -m|--mbr)
+            shift
+            MBR=true
+            NTFS=true
+            ;;
+        -w|--writable)
+            shift
+            WRITABLE=true
+            ;;
+        -f|--force)
+            shift
+            FORCE=true
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+if [ $# -lt 1 ] ; then
+    echo "Missing DEVICE" >&2
+    usage >&2
+    exit 1
+fi
+
+DEVICE="$1"
+
+USERID=$(id -u)
+if [ "$USERID" != "0" ]; then
+    echo "Program requires superuser privileges" >&2
+    exit 1
+fi
+
+# Check for required tools
+if $NTFS; then
```

Comment:
As mentioned in the other comment, I `NTFS` is controlled and using it as a boolean is entirely appropriate to me.